### DISCUSSION TITLE
context.WithTimeout cause memory leak?

### DISCUSSION THREAD

**Discussion by novrain** [yuin/gopher-lua#484] [Q&A]
Version v1.1.1
In my application, when i using context.WithTimeout to ensure timeoute, lua realted memory keep rising.
![屏幕截图 2024-03-22 143404](https://github.com/yuin/gopher-lua/assets/4167084/e0313007-8da3-4586-b115-289d22d09aa2)

When using context.WithTimeout, PCall terminated by exception, then how memory in this LuaState been GC?

**Comment by tul** [comment]
Just a thought - are you remembering to call the `cancel()` function that `context.WithTimeout()` returns?
